yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -353,7 +353,7 @@ private[yarn] class YarnAllocator(
 
     } else if (missing < 0) {
       val numToCancel = math.min(numPendingAllocate, -missing)
-      logInfo(s"Canceling requests for $numToCancel executor containers")
+      logInfo(s"Canceled requests for $numToCancel executor container(s)")
```

Comment:
Thanks for the explanation, I see what's happening there now. It would be more clear if we would add the target number to the cancel message like this:
`Canceling requests for $numToCancel executor container(s) to have a new desired total $targetNumExecutors executor(s).`